### PR TITLE
Add Foldable instance for Tree

### PR DESCRIPTION
**Pull request by hverr**
- [x] includes tests
- [x] ready for review
- [ ] reviewed by @skeuchel 

#### What does this PR do?
Implement a`Foldable` instance for `Tree`

### CODE REVIEW

File: tests/Properties.hs [hunk 1]
```diff
@@ -51,6 +54,10 @@ prop_fromSingletonIndex_singletonIndex :: Int64 -> Bool
 prop_fromSingletonIndex_singletonIndex i =
     fromSingletonIndex (singletonIndex i) == Just i
 
+prop_foldable :: [(Int64, Int)] -> Bool
+prop_foldable xs = F.foldMap snd xs' == F.foldMap id (Tree.fromList xs')
+  where xs' = nubBy (\x y -> fst x == fst y) . map (\x -> (fst x, Sum $ snd x)) $ xs
+
```
Comment (skeuchel) [inline tests/Properties.hs]: can you add this test case:

prop_toList_fromList :: [(Int64, Int)] -> Bool
prop_toList_fromList xs =
    F.toList (Tree.fromList xs) ==
    F.toList (M.fromList xs)

and if this fails, change the order in Tree.fromList

File: src/Data/BTree/Pure.hs [hunk 2]
```diff
@@ -233,6 +233,22 @@ findWithDefault v k = fromMaybe v . lookup k
...
 --------------------------------------------------------------------------------
 
+-- | Make a tree node foldable over its value.
+newtype FoldableNode height key val = FoldableNode (Node key val height)
+
+instance F.Foldable (Tree key) where
+    foldMap _ (Tree Nothing) = mempty
+    foldMap f (Tree (Just n)) = F.foldMap f (FoldableNode n)
+
+instance F.Foldable (FoldableNode height key) where
```
Comment (hverr) [inline src/Data/BTree/Pure.hs]: no counter argument, changing Node type seems like the best option

File: tests/Properties.hs (hunk 1 above)
Comment (hverr) [inline tests/Properties.hs]: will do

File: src/Data/BTree/Pure.hs (hunk 2 above)
Comment (hverr) [inline src/Data/BTree/Pure.hs]: see e0c912a

File: tests/Properties.hs (hunk 1 above)
Comment (hverr) [inline tests/Properties.hs]: see a6e92d7

**Comment by skeuchel** [comment]
LGTM

**Pull request merged**